client/src/util/metrics/parse.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.
+ *
+ * Camunda licenses this file to you under the MIT; you may not use this file
+ * except in compliance with the MIT License.
+ */
+
+import { is } from 'bpmn-js/lib/util/ModelUtil';
+
+import BpmnModdle from 'bpmn-moddle';
+
+import CamundaBpmnModdle from 'camunda-bpmn-moddle/resources/camunda';
+
+
+export async function getDefinitions(xml) {
+  const moddle = new BpmnModdle({
+    camunda: CamundaBpmnModdle,
+  });
+
+  const { rootElement: definitions } = await moddle.fromXML(xml);
+
+  return definitions;
+}
+
+function extractFlowElementsOfType(rootElement, type, flowElements) {
```

Comment:
We have [helper functions](https://github.com/bpmn-io/extract-process-variables/blob/master/src/util/ElementsUtil.js#L75) that solve this recursion problem for us. I wonder if we could re-use that.

```js
import { filter } from 'min-dash';

// collect all elements in a flat array
const elements = selfAndAllFlowElements([ rootElement ], false);

return filter(elements, (element) => {
  return is(element, type);
});
```